### DISCUSSION TITLE
Is there a way to set a TypeRegistrar on a CommandAppTester

### DISCUSSION THREAD

**Discussion by nils-a** [spectreconsole/spectre.console#461] [Q&A] [answered]
I have a setup using a registrar, for dependency injection.
So my `main` method looks like this:

```cs
private static int Main(string[] args)
{
  var registrar = BuildRegistrar();
  var app = new CommandApp(registrar);
  app.Configure(ConfigureApp);
  return app.Run(args);
}
```

Now, I'd like to test some of my commands using the `CommandAppTester`.
This however, seems impossible because dependencies do not get resolved.

Am I missing something?

**Comment by patriksvensson** [comment, marked as answer]
@nils-a I took a quick look at it and you are correct. Should be easily fixed though in the `CommandAppTester` if we add a `ITypeRegistrar` property which we later inject into the created `CommandApp`. 

Would you want to submit a PR for this? Otherwise I might be able to do it later this week.

**Reply by nils-a** [reply]
Already on it 😄

**Reply by nils-a** [reply]
I've opened https://github.com/spectreconsole/spectre.console/pull/462 - but I'm not 100% happy with it's unit test.
